### PR TITLE
Announcements

### PR DESCRIPTION
**Pull request by lightvector**
This adds a simple mechanism to add different sections to the front page with arbitrary html, from the admin page. Also it adds an acknowledgments line that we will use in the footer for KataGo's public run.

### CODE REVIEW

**Comment by lightvector** [comment]
Also added a test.

File: src/apps/announcements/apps.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from django.apps import AppConfig
+from django.utils.translation import gettext_lazy as _
+
+
+class AnnouncementsConfig(AppConfig):
+    """
+    Katago Announcments app handles some messages for the front page of the site
```
Comment (tychota) [inline src/apps/announcements/apps.py:7]: Nits: Typo "Announc**e**m**e**nts"

File: src/apps/announcements/admin/announcement_admin.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+class AnnouncementAdmin(admin.ModelAdmin):
+    """
+    AnnouncementAdmin allows admin to create or edit Announcements
+    """
+
+    list_display = (
+        "id",
+        "title",
+        "display_order",
```
Comment (tychota) [inline src/apps/announcements/admin/announcement_admin.py:13]: nice the display order.

File: src/apps/announcements/admin/announcement_admin.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        "enabled",
```
Comment (tychota) [inline src/apps/announcements/admin/announcement_admin.py:14]: And nice the enabled, would have done exactly the same.
:)

File: src/apps/announcements/models/announcement.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    title = CharField(
+        _("title"),
+        null=False,
+        blank=False,
+        max_length=240,
+        help_text=_("Title of the announcement section."),
+        db_index=True,
+        unique=True,
+    )
+    contents = TextField(
```
Comment (tychota) [inline src/apps/announcements/models/announcement.py:35]: Did you prefer raw html as something like markdown ?

Easier to code, but maybe a bit less clear than md to write.
Plus security breach vector if at some points people other than me or you can write annoncement (stored XSS).

File: src/apps/announcements/models/announcement.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    id = BigAutoField(primary_key=True)
+    created_at = DateTimeField(_("creation date"), auto_now_add=True, db_index=True)
+    updated_at = DateTimeField(_("updated date"), auto_now=True, db_index=True)
+    title = CharField(
+        _("title"),
+        null=False,
+        blank=False,
+        max_length=240,
+        help_text=_("Title of the announcement section."),
+        db_index=True,
```
Comment (tychota) [inline src/apps/announcements/models/announcement.py:32]: Do you really need an index here ?

**Review by tychota — APPROVED** [review]

File: src/apps/announcements/tests/test_announcements.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import pytest
+import base64
+
+from django.test import Client
+from src.apps.announcements.models import Announcement
+
+pytestmark = pytest.mark.django_db
+
+class TestAnnouncements:
```
Comment (tychota) [inline src/apps/announcements/tests/test_announcements.py:9]: nice

**Comment by tychota** [comment]
Also do we aims to translate the website at some point (eg make a chineese version) ? 
Everything should be nice and easy to translate (thanks django po/mo system and {trans} balises) but the new announcement doesn't include anything to say "this blog post is in english and shall appaer only on english pages".

File: src/apps/announcements/models/announcement.py (hunk 4 above)
Comment (lightvector) [reply src/apps/announcements/models/announcement.py:36]: Thanks, made it markdown. But I still only intend admins to write. :)

File: src/apps/announcements/models/announcement.py (hunk 5 above)
Comment (lightvector) [reply src/apps/announcements/models/announcement.py]: Unique probably implies index practically, but I removed it to just go with the default, thanks.

File: src/apps/announcements/apps.py (hunk 1 above)
Comment (lightvector) [reply src/apps/announcements/apps.py]: fixed

**Comment by lightvector** [comment]
Going ahead to merge this, but I would be interested to hear your thoughts for how to produce a translated version. Thanks!

**Pull request merged**